### PR TITLE
StorageSystem creation wizard: fix replicas calculation

### PR DESCRIPTION
**Pull request by alfonsomthd**
- Calculate the amount of replicas taking into account zones, racks and the minimum replicas.
- Added unit tests.

### CODE REVIEW

**Comment by SanjalKatiyar** [comment]
/retitle StorageSystem creation wizard: fix replicas calculation
now we have 4.13 branches, we need to add BZ number to the title for backport PRs now... not for master...

**Comment by SanjalKatiyar** [comment]
> /retitle StorageSystem creation wizard: fix replicas calculation now we have 4.13 branches, we need to add BZ number to the title for backport PRs now... not for master...

once backport PRs are up (after this master PR is merged), add the BZ number to the title of both PRs and make sure both PRs are added to "Links" section of the BZ... after that we can move BZ to "POST" and need to ask for "qa_ack" on the BZ... once done we can merge the backport PRs and move BZ to MODIFIED...

**Review by bipuladh — APPROVED** [review]

File: packages/odf/components/utils/common.ts [hunk 1]
```diff
@@ -292,28 +296,51 @@ export const prettifyJSON = (data: string) =>
...
+  }, new Set<string>());
+};
+
+export const getReplicasFromSelectedNodes = (
+  nodes: WizardNodeState[]
+): number => {
+  const zones = getUniqueZonesSet(nodes);
+  let replicas = zones.size;
+  // If there are no zones, look for racks.
+  if (replicas === 1 && zones.has(undefined)) {
```
[the remark this answers is not in the document]
Comment (alfonsomthd) [reply packages/odf/components/utils/common.ts:319]: @SanjalKatiyar when there are no zones, it returns a set that contains `undefined` so it ends having 1 replica... As I wasn't sure if this was intended, I didn't modify the behavior of `getUniqueZonesSet`. IF we agree on adjusting the return values of the aforementioned function, I can refactor accordingly.

[the remark this answers is not in the document]
Comment (SanjalKatiyar) [reply packages/odf/components/utils/common.ts:319]: my bad... its alright !

**Pull request merged**